Utils/Dataflow/009_oracleConnector/Oracle2JSON.py

```diff
@@ -0,0 +1,307 @@
+            n = join_buffer.get(tid, 0) + 1
+            if n == req_n:
+                yield d[tid]
+                del join_buffer[tid]
+                del d[tid]
+            else:
+                join_buffer[tid] = n
+
+    # Flush records stuck in the buffer (not joined)
+    for tid in d:
+        yield d[tid]
+
+
+def process(conn, offset_date, final_date_cfg, step_seconds, queries):
+    if final_date_cfg:
+        final_date = final_date_cfg
+    else:
+        final_date = date2str(datetime.now())
+    break_loop = False
+    while (not break_loop and str2date(offset_date) < str2date(final_date)):
+        end_date = date2str(str2date(offset_date)
+                            + timedelta(seconds=step_seconds))
+        if str2date(end_date) > str2date(final_date):
+            end_date = final_date
+            break_loop = True
+        sys.stderr.write("(TRACE) %s: Run queries for interval from %s to %s\n"
+                         % (date2str(datetime.now()), offset_date, end_date))
+        if mode == SQUASH_POLICY:
+            records = squash(conn, queries, offset_date, end_date)
+        elif mode == PLAIN_POLICY:
+            records = plain(conn, queries, offset_date, end_date)
+        for r in records:
+            r['phys_category'] = get_category(r)
+            sys.stdout.write(json.dumps(r) + '\n')
+        update_offset(end_date)
+        offset_date = end_date
+        if not final_date_cfg:
+            final_date = date2str(datetime.now())
+
+def get_initial_date():
+    """
+     TODO:
+        Procedure to obtain the min(timestamp) from t_production_task
+        SELECT min(timestamp) from t_production_task;
+    """
+    return None
+
+def query_executor(conn, sql_file, offset_date, end_date):
+    """
+    Execution of query with offset from file
+    """
+    try:
+        file_handler = open(sql_file)
+        query = file_handler.read().rstrip().rstrip(';') % (offset_date, end_date)
+        return DButils.ResultIter(conn, query, 1000, True)
+    except IOError:
+        sys.stderr.write('File open error. No such file (%s).\n' % sql_file)
+        sys.exit(2)
+
+
+def get_offset():
+    config = ConfigParser.ConfigParser()
+    config.read(conf)
+    return config.get("timestamps", "offset")
+
+def update_offset(new_offset):
+    """
+    Updating offset value in configuration file
+    """
+    config = ConfigParser.RawConfigParser()
+    config.optionxform = str
+    config.read(conf)
+    config.set('timestamps', 'offset', new_offset)
+    with open(conf, 'w') as configfile:
+        config.write(configfile)
+
+def interval_seconds(step):
+    """
+    Convert human-readable interval into seconds.
+
+    If no suffix present, take step as seconds.
+    """
+    try:
+        return int(step)
+    except ValueError:
+        pass
+    if len(step) < 2:
+        raise ValueError("Failed to decode interval: %s" % step)
+    suffix = { 'd': 86400, 'h': 3600, 'm': 60, 's': 1}
+    val = step[:-1]
```

Comment:
If step just 10 (without suffix, to pass length check), then **val** will be just 1.